python/cudf_polars/tests/expressions/test_numeric_unaryops.py

```diff
@@ -112,3 +116,42 @@ def test_null_count():
         pl.col("baz").is_null().sum(),
     )
     assert_gpu_result_equal(q)
+
+
+@pytest.mark.parametrize("method", ["ordinal", "dense", "min", "max", "average"])
+@pytest.mark.parametrize("descending", [False, True])
+def test_rank_supported(request, ldf: pl.LazyFrame, method: str, *, descending: bool):
+    request.applymarker(
+        pytest.mark.xfail(condition=POLARS_VERSION_LT_132, reason="nested loop join")
+    )
+    expr = pl.col("a").rank(method=method, descending=descending)
+    q = ldf.select(expr)
+    assert_gpu_result_equal(q)
+
+
+@pytest.mark.parametrize("method", ["ordinal", "dense", "min", "max", "average"])
+@pytest.mark.parametrize("descending", [False, True])
+@pytest.mark.parametrize("test", ["with_nulls", "with_ties"])
+def test_rank_methods_with_nulls_or_ties(
+    request, ldf: pl.LazyFrame, method: str, *, descending: bool, test: str
+) -> None:
+    request.applymarker(
+        pytest.mark.xfail(condition=POLARS_VERSION_LT_132, reason="nested loop join")
+    )
+
+    base = pl.col("a")
+    if test == "with_nulls":
+        expr = pl.when((base % 2) == 0).then(None).otherwise(base)
+    else:
+        expr = pl.when((base % 2) == 0).then(pl.lit(-5)).otherwise(base)
+
+    q = ldf.select(expr.rank(method=method, descending=descending))
+    assert_gpu_result_equal(q)
+
+
+@pytest.mark.parametrize("seed", [42])
+@pytest.mark.parametrize("method", ["random"])
+def test_rank_unsupported(ldf: pl.LazyFrame, method: str, seed: int) -> None:
+    expr = pl.col("a").rank(method=method, seed=seed)
```

Comment:
Minor, but IIUC seed is only really relevant for `method="random"` and so having it as an independent parameter from `method` maybe isn't appropriate

```suggestion
@pytest.mark.parametrize("method_kwargs", [("kwargs", {"seed": 42})])
def test_rank_unsupported(ldf: pl.LazyFrame, method_kwargs: tuple) -> None:
    method, kwargs = method_kwargs
    expr = pl.col("a").rank(method=method, **kwargs)
```

But feel free to ignore, since this is the only method we don't (currently) support.